### PR TITLE
Export msys2 based environment on windows

### PR DESCRIPTION
**Pull request by ksubrama**
Needs https://github.com/chef-cookbooks/build-essential/pull/111 and https://github.com/chef-cookbooks/mingw/pull/6

### CODE REVIEW

**Comment by ksubrama** [comment]
@tyler-ball @btm

**Comment by btm** [comment]
:+1:

File: recipes/_compile.rb [hunk 1]
```diff
@@ -80,12 +80,9 @@
...
-                 node['build-essential']['mingw32']['path']
-               else
-                 node['build-essential']['mingw64']['path']
-               end
-
-  omnibus_env['PATH'] << windows_safe_path_join(mingw_path, 'bin')
-  omnibus_env['PATH'] << windows_safe_path_join(mingw_path, 'msys', '1.0', 'bin')
+  bitness = windows_arch_i386? ? '32' : '64'
+  msys_path = node['build-essential']['msys64']['path']
+  omnibus_env['MSYSTEM'] << "MINGW#{bitness}"
```
Comment (schisamo) [inline recipes/_compile.rb]: Is `MSYSTEM` an environment variable we made up? Are `MINGW32` and `MINGW64` both valid values?

Comment (btm) [inline recipes/_compile.rb]: No, `MSYSTEM` is an upstream variable, for example https://sourceforge.net/p/msys2/tickets/4/.

**Comment by mwrock** [comment]
:+1: after you make up with travis

File: recipes/_user.rb [hunk 2]
```diff
@@ -44,10 +44,13 @@
...
 end
 
-group node['omnibus']['build_user_group'] do
+group "create #{node['omnibus']['build_user_group']} group" do
+  group_name node['omnibus']['build_user_group']
   # The Window's group provider get's cranky if attempting to create a
   # built-in group.
   ignore_failure true if windows?
+  # Don't remove existing members - in case this is a built-in group.
+  append true
```
Comment (tyler-ball) [inline recipes/_user.rb]: Does this fix the error about the `vagrant` group when I'm running in Test Kitchen?

**Comment by tyler-ball** [comment]
👍 Ditto to Matt

File: recipes/_user.rb (hunk 2 above)
Comment (ksubrama) [inline recipes/_user.rb]: I think it's slightly weirder.  append true when used with modify doesn't delete existing members from a group but doesn't create the group if it doesn't exist.  append true when used with create will not remove existing members from the group if it doesn't exist but the windows provider is whacky and errors if you try to create a preexisting group (like Administrators).  So we attempt to create the group while ignoring errors and then to add to the group.

The weird error you were seeing before is due to the fact that we had two resources that were named the same thing - and so their properties got cloned from one to the other and hilarious bad things happened but the error was being ignored.

File: Berksfile [hunk 3]
```diff
@@ -6,6 +6,10 @@ source 'https://supermarket.chef.io'
 
 metadata
 
+# Just for testing
+cookbook 'mingw', github: 'chef-cookbooks/mingw', branch: 'msys2'
```
Comment (tyler-ball) [inline Berksfile]: This got merged, no longer need to point at this branch

**Pull request merged**